src/DenoHTTPWorker.ts

```diff
@@ -338,56 +336,54 @@ class denoHTTPWorker {
-    options: http.RequestOptions,
-    callback: (response: http.IncomingMessage) => void
-  ): http.ClientRequest {
-    options.headers = (options.headers || {}) as http.OutgoingHttpHeaders;
-
-    // TODO: ensure these are handled with the correct casing?
-    delete options.headers["x-deno-worker-host"];
-    delete options.headers["x-deno-worker-connection"];
-
-    // NodeJS will send both the host and the connection headers
-    // (https://nodejs.org/api/http.html#new-agentoptions). We don't want these
-    // to make it to Deno unless they are explicitly set by the user. So store
-    // them to reconstruct on the other size.
-    if (options.headers.host) {
-      options.headers["X-Deno-Worker-Host"] = options.headers.host;
+  async request(options: RequestOptions): Promise<ResponseData> {
+    // UndiciHeaders can be an array or an object, so we normalize so we can
+    // manipulate the header options.
+
+    const headers = options.headers || new Headers();
+
+    headers.set("x-deno-worker-url", options.url);
+
+    // To prevent the user from setting these headers, we either update them to
+    // the real host / connection, or clear them
+    headers.delete("x-deno-worker-host");
+    headers.delete("x-deno-worker-connection");
+
+    const host = headers.get("host");
+    if (host) {
+      headers.set("x-deno-worker-host", host); // overwrites if set: https://developer.mozilla.org/en-US/docs/Web/API/Headers/set
     }
-    if (options.headers.connection) {
-      options.headers["X-Deno-Worker-Connection"] = options.headers.connection;
+
+    const connection = headers.get("connection");
+    if (connection) {
+      headers.set("x-deno-worker-connection", connection);
     }
 
-    options.headers = {
-      ...options.headers,
-      "X-Deno-Worker-URL": typeof url === "string" ? url : url.toString(),
-    };
-    url = "http://deno";
-    options.agent = this.#agent;
-    options.socketPath = this.#socketFile;
-    return http.request(url, options, callback);
+    const resp = await this.#pool.request({
+      ...options,
+      origin: "http://deno",
+      path: "/",
+      query: {},
+      headers,
+    });
+
+    return {
+      statusCode: resp.statusCode,
+      headers: resp.headers,
+      body: resp.body,
+      trailers: resp.trailers,
+      context: resp.context || {},
+    }
   }
 
   // We send this request to Deno so that we get a live connection in the
   // http.Agent and subsequent requests are do not have to wait for a new
   // connection.
   async warmRequest() {
-    return new Promise<void>((resolve, reject) => {
-      const req = http.request(
-        "http://deno",
-        { agent: this.#agent, socketPath: this.#socketFile },
-        (resp) => {
-          resp.on("error", reject);
-          resp.on("data", () => {});
-          resp.on("close", () => {
-            resolve();
-          });
-        }
-      );
-      req.on("error", reject);
-      req.end();
+    return this.#pool.request({
```

Comment:
```suggestion
    return await this.#pool.request({
```

Using await here to add `warmRequest` to a potential async stacktrace